### PR TITLE
Update CHANGELOG

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## [1.2](https://github.com/arborx/arborx/tree/v1.2) (2022-03-??)
```
Comment (aprokop) [inline CHANGELOG.md:3]: FIXME

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+- Allow alternate space filling curves (EXPERIMENTAL) [\#646](https://github.com/arborx/ArborX/pull/646)
+- Remove certain helper functions from the `ArborX` namespace [\#648](https://github.com/arborx/ArborX/pull/648)
+
+**Fixed bugs:**
+
+- Enable testing when ARBORX_ENABLE_BENCHMARKS is ON [\#555](https://github.com/arborx/ArborX/pull/555)
+- Fix the exit status of the DBSCAN example [\#566](https://github.com/arborx/ArborX/pull/566)
+- Bring DBSCAN README documentation in line with the code [\#568](https://github.com/arborx/ArborX/pull/568)
+- Fix a race condition in the BVH construction [\#579](https://github.com/arborx/ArborX/pull/579)
+- Fix a bug in the nearest search using the DistributedTree [\#653](https://github.com/arborx/ArborX/pull/653)
```
Comment (dalg24) [inline CHANGELOG.md:37]: and improved performance

Comment (aprokop) [reply CHANGELOG.md:37]: Not really, we never had any performance in the first place, because it was not working correctly.

Comment (dalg24) [reply CHANGELOG.md:37]: Lower memory footprint, one less kernel launch

Comment (aprokop) [reply CHANGELOG.md:37]: Too much info IMHO. Fell free to update.

**Comment by aprokop** [comment]
I'd like to get #661 merged prior to making the release. But that's the only thing.

**Comment by aprokop** [comment]
The CHANGELOG is ready.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+- Require Kokkos 3.4 [\#578](https://github.com/arborx/ArborX/pull/578)
+
+**Enhancements:**
+
+- Indicate the status of the worktree in the hash string ("-dirty") [\#558](https://github.com/arborx/ArborX/pull/558)
+- Check valid callback and space accessibility [\#563](https://github.com/arborx/ArborX/pull/563)
+- Add a molecular dynamics example [\#564](https://github.com/arborx/ArborX/pull/564)
+- Add nightly testing against Kokkos develop [\#569](https://github.com/arborx/ArborX/pull/569)
+- Improve performance of the HIP backend when using HIP-Clang [\#575](https://github.com/arborx/ArborX/pull/575)
+- Remove PGI CI [\#590](https://github.com/arborx/ArborX/pull/590)
```
Comment (dalg24) [inline CHANGELOG.md:22]: Do you really want to mention this?  Also why would you list it under enhancement?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,46 @@
 # Changelog
 
+## [1.2](https://github.com/arborx/arborx/tree/v1.2) (2022-04-01)
+
+[Full Changelog](https://github.com/arborx/arborx/compare/v1.1...v1.2)
+
+**New features:**
+
+- Implement Euclidean minimum spanning tree algorithm [\#624](https://github.com/arborx/ArborX/pull/624), [\#631](https://github.com/arborx/ArborX/pull/631), [\#661](https://github.com/arborx/ArborX/pull/631)
```
Comment (dalg24) [inline CHANGELOG.md:9]: Should we say it it experimental?  It is defined in `ArborX::Details::` which usually signify you are not allowed to use it.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,46 @@
...
+- Remove certain helper functions from the `ArborX` namespace [\#648](https://github.com/arborx/ArborX/pull/648)
+- Add ray-triangle intersection algorithm [\#617](https://github.com/arborx/ArborX/pull/617)
+
+**Deprecations:**
+
+- Deprecate `ArborX::InlineCallbackTag` [\#656](https://github.com/arborx/ArborX/pull/656)
+
+**Fixed bugs:**
+
+- Enable testing when ARBORX_ENABLE_BENCHMARKS is ON [\#555](https://github.com/arborx/ArborX/pull/555)
```
Comment (dalg24) [inline CHANGELOG.md:38]: Would not mention it

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,46 @@
...
+- Add nightly testing against Kokkos develop [\#569](https://github.com/arborx/ArborX/pull/569)
+- Improve performance of the HIP backend when using HIP-Clang [\#575](https://github.com/arborx/ArborX/pull/575)
+- Remove PGI CI [\#590](https://github.com/arborx/ArborX/pull/590)
+- Add Point-Sphere intersection algorithm [\#584](https://github.com/arborx/ArborX/pull/584)
+- Avoid host copy in distributed tree construction when using CUDA-aware MPI [\#597](https://github.com/arborx/ArborX/pull/597)
+- Print Kokkos version during configuration [\#609](https://github.com/arborx/ArborX/pull/609)
+- Improve performance of the brute force algorithm [\#616](https://github.com/arborx/ArborX/pull/616)
+- Use 64-bit Morton indices by default in the construction [\#637](https://github.com/arborx/ArborX/pull/637)
+- Allow alternate space filling curves (EXPERIMENTAL) [\#646](https://github.com/arborx/ArborX/pull/646)
+- Remove certain helper functions from the `ArborX` namespace [\#648](https://github.com/arborx/ArborX/pull/648)
```
Comment (dalg24) [inline CHANGELOG.md:29]: This falls under deprecation, not enhancement

File: CHANGELOG.md (hunk 3 above)
Comment (masterleinad) [reply CHANGELOG.md:22]: I wouldn't argue that it is an enhancement with respect to maintainability to drop it. :wink: Seriously, I would either list it under "incompatibilities" or drop it.

File: CHANGELOG.md (hunk 4 above)
Comment (masterleinad) [reply CHANGELOG.md:9]: I think it would be better fit in `namespace Experimental` than `namespace Details` then.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,46 @@
...
+**Build changes:**
+
+- Require Kokkos 3.4 [\#578](https://github.com/arborx/ArborX/pull/578)
+
+**Enhancements:**
+
+- Indicate the status of the worktree in the hash string ("-dirty") [\#558](https://github.com/arborx/ArborX/pull/558)
+- Check valid callback and space accessibility [\#563](https://github.com/arborx/ArborX/pull/563)
+- Add a molecular dynamics example [\#564](https://github.com/arborx/ArborX/pull/564)
+- Add nightly testing against Kokkos develop [\#569](https://github.com/arborx/ArborX/pull/569)
```
Comment (masterleinad) [inline CHANGELOG.md:20]: Not quite sure if we should mention CI changes at all.

File: CHANGELOG.md (hunk 6 above)
Comment (masterleinad) [reply CHANGELOG.md:29]: It's also pretty vague and thus doesn't sound significant.

File: CHANGELOG.md (hunk 4 above)
Comment (aprokop) [reply CHANGELOG.md:9]: I would not worry about what namespace it live in right now.

File: CHANGELOG.md (hunk 3 above)
Comment (aprokop) [reply CHANGELOG.md:22]: Dropped.

File: CHANGELOG.md (hunk 6 above)
Comment (aprokop) [reply CHANGELOG.md:29]: Clarified and moved to deprecated.

File: CHANGELOG.md (hunk 5 above)
Comment (aprokop) [reply CHANGELOG.md:38]: OK.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**